src/lib/application.ts

```diff
@@ -91,8 +91,8 @@ export class Application extends ChildableComponent<Application, AbstractCompone
         super(null);
 
         this.logger    = new ConsoleLogger();
-        this.converter = this.addComponent('converter', Converter);
-        this.renderer  = this.addComponent('renderer', Renderer);
+        this.converter = this.addComponent<Converter>('converter', Converter);
+        this.renderer  = this.addComponent<Renderer>('renderer', Renderer);
```

Comment:
I'm curious why these components need to be explicitly typed but the PluginHost and Options do not

Reply:
It was a change done in 2.5 that I wasn't able to pin down. I'm not familiar enough with the Typedoc codebase.

Reply:
Looks like the error is

> Type 'AbstractComponent<Application> & Component' is not assignable to type 'Converter'.
  Property 'name' is missing in type 'AbstractComponent<Application> & Component'.

So I think the error is due to a change in TypeScript type inference. I didn't dig deep enough into the class inheritance to figure out why the other components are inferred/casted differently.